realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -998,16 +1015,21 @@ public void removeChangeListeners() {
      * Notifies all registered listeners.
      */
     void notifyChangeListeners() {
```

Comment:
in the `HandlerController` you need to call the overload one with `syncBeforeNotifiying == false` here example: https://github.com/realm/realm-java/blob/cm/bug/iterators/realm/realm-library/src/main/java/io/realm/HandlerController.java#L403